GASocialLogin/Google/GAGoogleLoginService.swift

```diff
@@ -68,107 +75,138 @@ extension GASocialLogin
-            googleSignIn?.shouldFetchBasicProfile = true
-            googleSignIn?.delegate = self
-            googleSignIn?.presentingViewController = parentViewController
-            googleSignIn?.restorePreviousSignIn()
+            intiateSignInFlow(using: .silent, viewController: viewController, willDispatchHandler: willDispatchHandler, completion: completion)
         }
         
         /// Call to google signOut, marks current user as being in the signed out state.
         public func signOut()
         {
-            let googleSignIn = GIDSignIn.sharedInstance()
-            googleSignIn?.shouldFetchBasicProfile = true
-            googleSignIn?.delegate = self
-            googleSignIn?.presentingViewController = parentViewController
-            googleSignIn?.signOut()
+            googleSignIn.signOut()
         }
         
         /// Call to google disconnect, disconnects the current user from the app and revokes previous authentication. If the operation
         /// succeeds, the OAuth 2.0 token is also removed from keychain.
         public func disconnect()
         {
-            let googleSignIn = GIDSignIn.sharedInstance()
-            googleSignIn?.shouldFetchBasicProfile = true
-            googleSignIn?.delegate = self
-            googleSignIn?.presentingViewController = parentViewController
-            googleSignIn?.disconnect()
+            googleSignIn.disconnect()
         }
         
-        private func cleanBlocks()
+        // MARK: - Private API Method
+        /// Call to google signInSilently/signIn ,
+        /// Attempts to sign in a previously authenticated user without interaction.  The closure will be
+        /// called at the end of this process indicating success or failure.
+        ///
+        /// - Parameters:
+        ///   - method: the sign in method (silent,non-silent)
+        ///   - viewController: the current present view controller
+        ///   - willDispatchHandler: block to handle signinWillDispatch
+        ///   - successHandler: log in results call back
+        private func intiateSignInFlow(using method:GIDSignInMethod, viewController: UIViewController, willDispatchHandler: GAGoogleWillDispatchBlock? = nil, completion: @escaping GAGoogleCompletion)
         {
-            googleCompletion        = nil
-            googleWillDispatchBlock = nil
+            parentViewController    = viewController
+            googleCompletion        = completion
+            googleWillDispatchBlock = willDispatchHandler
+            //can be removed and used the strong reference or should i use the weak property?
+            guard let parentViewController = parentViewController else
+            {
+                let userInfo    = [
+                    NSLocalizedDescriptionKey : "\(#function) : viewController passed in was nil."
+                ]
+                let error       = NSError(domain: GAGoogleLoginService.serviceKey, code: 170, userInfo: userInfo)
```

Comment:
@lidorf-gini-apps why do you use 170 as the code?

Reply:
its been a long time..
As i recall the code was chosen randomly (error codes can be any number),
what number will work better?